addon/services/cookies.js

```diff
@@ -212,5 +215,14 @@ export default Service.extend({
     }
 
     return cookie;
-  }
+  },
+    
+  _isCookieSizeAcceptable(value) {
+    return this._getByteCount(value) < MAX_COOKIE_BYTE_LENGTH;
+  },      
+    
+  _getByteCount(value) {
+    return typeof(value) === 'string' ? encodeURI(value).split(/%(?:u[0-9A-F]{2})?[0-9A-F]{2}|./).length - 1 : 0;
```

Comment:
When the `value` assigned in line 82 is passed here, the `typeof` check would no longer be necessary.

Also it would be good to add a comment explaining what the `split` does.